### ISSUE TITLE
Showing only available days

### ISSUE THREAD

**Issue by agorkem** [Hacker0x01/react-datepicker#195]
There is `excludeDates` property for listing excluded dates but I need to show only available days. I need something like that:

```js
<DatePicker
    selected={this.state.startDate}
    onChange={this.handleChange}
    availableDates={[moment()...]}
    placeholderText="Select a date other than today or yesterday" />
```

Is there any better way to do that with **current** props? If there is no I think it's easy to add this feature.

**Comment by martijnrusschen** [comment]
Does a date range work for you? http://hacker0x01.github.io/react-datepicker/ has an example that only enables a specific date range. If not, it should be able to build something that fits your needs eaisly. The only problem is that I don't have the time to build it in the next few weeks, so if you can submit a pull request, I'm happy to review it.

**Comment by timukasr** [comment]
I have a similar request: I want to whitelist/disable dates by a function. `excludeDates` or some other prop could accept a function, which takes a Date object as argument and returns true/false to enable/disable a date.

One use case is to disable all weekends. Currently I don't see a good solution (besides generating very large `excludeDates` array and hoping users don't navigate too far away).

**Comment by martijnrusschen** [comment]
We could consider something like https://www.npmjs.com/package/moment-business to detect weekends?

**Comment by timukasr** [comment]
Detecting weekends is not the main point here. It's more like "disable/enable days based on a recurring rule". If we have `minDate` and `maxDate`, we could generate all the days and use the current `excludeDate` prop. But with infinite range we cannot use this approach.

Weekends was just one example. My current problem is to "disable all the weekends and national holidays". Hardcoding this to react-datepicker is not feasible (due to localities and whatnot) and also it is not very dynamic (others might have different rules to disable/enable days).

**Comment by martijnrusschen** [comment]
I merged a new feature today that introduces includeDates. I guess that fixes this problem too. Likely to release this new feature tomorrow in a full packed release.

**Comment by martijnrusschen** [comment]
Fixed by #247

**Comment by jaypatel512** [comment]
@martijnrusschen ! I know I am commenting on a really old issue here, but felt as it was quite relevant. I am hoping to use this calendar to show the appointment calendar for a shop. 

They have 30 minutes slots that someone can register. I fetch the data from their server getting list of dates, and array of times that are available for that day. 

I tried using includeTime/includeDate, but using includeTime opens that slot for all days. Is there a way to just give them the raw input of dates (which includes time as well), and render only those as acceptable values. (Just like a normal appointment calendar) showing all available slots.
